org.openhab.binding.zigbee/src/main/resources/OH-INF/thing/xiaomi/xiaomi_lumisensor-switch.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="zigbee"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- These devices do not correctly report their clusters, so they need a static thing definition for the binding to inject the correct clusters into the ZigBeeNode -->
+
+	<thing-type id="xiaomi_lumisensor-switch" listed="false">
+		<label>Xiaomi LUMI Mijia Button</label>
+		<description>Xiaomi LUMI Mijia Button (round)</description>
+
+		<channels>
+			<channel id="switch" typeId="switch_onoff">
+				<properties>
+					<property name="zigbee_endpoint">1</property>
+					<property name="zigbee_inputclusters">6</property>
+				</properties>
+			</channel>
+		</channels>
+
+		<properties>
```

Comment:
I don't believe you need to add these properties - they are automatically added by the binding. I know they are also in some of the other files and I plan to remove these when I do some major refactoring to split the Tuya bundle out.

Reply:
Without it the button never gets the endpoint to register for me despite how many attempts or how long I hold down the link button.  Having this xml seems to pre-seed the device endpoint "1" as being known so it won't get stuck at with an unknown remote endpoint early return point in com.zsmartsystems.zigbee ZigBeeNetworkManager.java(line 1086) -- receiveZclCommand().

My other attempt (which also worked for me) was to change the ZigBeeNetworkManager to ignore that short circuit return and add the endpoint in that unknown situation but you indicated that would violate compliance to do it at that level.

Reply:
> Without it the button never gets the endpoint

Really? I'm surprised since this information is used to detect the device so adding it to the XML, which is used AFTER the device is detected, doesn't make sense. We don't normally add this information on other devices, although I appreciate it has crept in to some of the Xiaomi devices.

Just to be clear - the binding reads the properties such as the vendor name, and model, and logical type. It then sets these properties, and uses this information to decide the thing type. You adding these properties here can only be used AFTER the binding has already read them - before this, the XML is not used.

> Having this xml seems to pre-seed the device endpoint "1" as being known

Sure - but these properties (vendor, modeId and localtype) should not be related to the channel definition.

> My other attempt (which also worked for me) was to change the ZigBeeNetworkManager to ignore that short circuit return and add the endpoint in that unknown situation but you indicated that would violate compliance to do it at that level.

Absolutely - what you are doing here is the correct approach and I'm perfectly happy with the channel definition. I'm just surprised that it doesn't work with these properties added. I can't test this without the device, but these properties are not used in the binding so i really don't understand why you say they are required?

Reply:
I haven't 100% tracked it thru the codepath but I get the impression that the modelID property comes thru correctly during device detection but the endpoint/channel definitions don't.  I got this impression that this xml file gets applied after the modelId match is detected which then safelists the endpoint definition in the xml file (i.e., indicating that endponit 1 is valid) thus letting it get bypass the flawed portion of the endpoint/channel definition reporting from the device.

Reply:
> but the endpoint/channel definitions don't.

Sure - but I'm not asking you to remove the endpoint / channel definitions. I'm talking about this property section here that defines the model, vendor, and logical type. These are not used in the binding so I really can't understand how you say it doesn't work without them. Maybe you're right - but it doesn't make sense given these are not used - I'm just confused. Also if it was required, why do other definitions not have this.

Sorry - I'm just a bit confused about this and need to look through it to confirm this really isn't used, but from my quick search I can't see anything in the code.

Reply:
Ah. I understand now.   I'll see how much I can trim it back.  sorry for the spam.